extensions/dim-rename-kafka8/src/main/java/io/druid/server/namespace/KafkaExtractionManager.java

```diff
@@ -0,0 +1,231 @@
+import kafka.serializer.Decoder;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.regex.Pattern;
+
+/**
+ *
+ */
+@ManageLifecycle
+public class KafkaExtractionManager
+{
+  private static final Logger log = new Logger(KafkaExtractionManager.class);
+
+  private final Properties kafkaProperties;
+  private final Collection<ListenableFuture<?>> futures = new ConcurrentLinkedQueue<>();
+  private final ListeningExecutorService executorService = MoreExecutors.listeningDecorator(
+      Executors.newCachedThreadPool(
+          new ThreadFactoryBuilder()
+              .setNameFormat("kafka-rename-consumer-%d")
+              .setDaemon(true)
+              .setPriority(Thread.MIN_PRIORITY)
+              .build()
+      )
+  );
+  private final AtomicInteger startedPipes = new AtomicInteger(0);
+  private final AtomicLong events = new AtomicLong(0l);
+
+  // Bindings in KafkaExtractionNamespaceModule
+  @Inject
+  public KafkaExtractionManager(
+      Injector injector
+  )
+  {
+    if (!NamespacedExtractionModule.isNamespaceEnabled(injector)) {
+      this.kafkaProperties = null;
+    } else {
+      final Properties kafkaProperties = injector.getInstance(
+          Key.get(
+              Properties.class,
+              Names.named("renameKafkaProperties")
+          )
+      );
+      this.kafkaProperties = new Properties();
+      if (kafkaProperties != null) {
+        if (kafkaProperties.containsKey("group.id")) {
+          throw new IAE(
+              "Cannot set kafka property [group.id]. Property is randomly generated for you. Found [%s]",
+              kafkaProperties.getProperty("group.id")
+          );
+        }
+        if (kafkaProperties.containsKey("auto.offset.reset")) {
+          throw new IAE(
+              "Cannot set kafka property [auto.offset.reset]. Property will be forced to [smallest]. Found [%s]",
+              kafkaProperties.getProperty("auto.offset.reset")
+          );
+        }
+        this.kafkaProperties.putAll(kafkaProperties);
+      }
+      if (!this.kafkaProperties.containsKey("zookeeper.connect")) {
+        this.kafkaProperties.put("zookeeper.connect", "localhost:2181/kafka");
+      }
+      // Enable publish-subscribe
+      this.kafkaProperties.setProperty("auto.offset.reset", "smallest");
+    }
+  }
+
+  public long getBackgroundTaskCount()
+  {
+    return startedPipes.get();
+  }
+
+  private static final Decoder<String> defaultStringDecoder = new Decoder<String>()
+  {
+    @Override
+    public String fromBytes(byte[] bytes)
+    {
+      return StringUtils.fromUtf8(bytes);
+    }
+  };
+
+  public long getNumEvents()
+  {
+    return events.get();
+  }
+
+  private static void updateMap(
```

Comment:
I only see this being called from one location.  I'm not sure it's adding to the readability more than just having it inline?